app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java

```diff
@@ -2719,15 +2720,18 @@ protected ContentHandle canonicalizeContent(ArtifactType artifactType, ContentHa
         }
     }
 
-    protected EditableArtifactMetaDataDto extractMetaData(ArtifactType artifactType, ContentHandle content) {
+    protected EditableArtifactMetaDataDto extractMetaData(ArtifactType artifactType, ContentHandle content, EditableArtifactMetaDataDto metaData) {
+        if (metaData == null) {
+            metaData = new EditableArtifactMetaDataDto();
+        }
```

Comment:
It's not usually a good idea to do this, although it is convenient.  In this case I don't think we want to pass in the metaData to this method at all, so I would ask that you revert this change and we can keep calling this method only if no meta-data was provided by the user.

Reply:
+1